Storage/StorageManager.php

```diff
@@ -161,68 +118,177 @@ public function storeUploadedFile($type, UploadedFile $uploadedFile, $newFilenam
      */
     public function storeFileFromLocalPath($type, $path, $newFilename = null, $unlinkAfterStore = false)
     {
-        if (!file_exists($path)) {
-            throw new \InvalidArgumentException(sprintf("Cannot store file from non-existent path at '%s'", $path));
+        $options = [];
+        if ($newFilename !== null) {
+            $options['newFilename'] = $newFilename;
         }
-        $this->validateFileStorageType($type);
-
-        $storage = $this->getStorageFromWritePolicy($type);
+        if ($unlinkAfterStore !== false) {
+            $options['unlinkAfterStore'] = $unlinkAfterStore;
+        }
+        return $this->store($type, $path, $options);
+    }
 
-        if (null === $newFilename) {
-            $newFilename = uniqid('gen' . substr(hash('sha256', $path), 0, 4));
+    /**
+     * Takes a content and store it in permanent storage
+     * Content might be:
+     * UploadedFile
+     * string - path to local file
+     * StreamInterface - stream containing file contents
+     *
+     * It is the callers responsibility to associate the returned entity with a relation and save it to the database
+     *
+     * @param integer $type
+     * @param string $content
+     * @param array $options
+     * @throws \RuntimeException If writing the file fails
+     * @return StorageFile
+     */
+    public function store($type, $content, array $options = [])
```

Comment:
This function should be added to the interface, and the deprecated ones marked as such on the interface.

Reply:
Also, the function signature should document all available options.  Use a similar documentation method as the php aws sdk does for the options.

Reply:
Please use the OptionResolver component to validate options.